bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/discovery/BigAssFanDiscoveryService.java

```diff
@@ -115,21 +115,23 @@ protected void stopBackgroundDiscovery() {
         cancelListenerJob();
     }
 
-    private void startListenerJob() {
-        if (listenerJob == null) {
+    private synchronized void startListenerJob() {
+        logger.debug("Starting discovery listener job in {} seconds", BACKGROUND_DISCOVERY_DELAY);
+
+        if (this.listenerJob == null) {
             terminate = false;
-            logger.debug("Starting discovery listener job in {} seconds", BACKGROUND_DISCOVERY_DELAY);
-            listenerJob = scheduledExecutorService.schedule(listenerRunnable, BACKGROUND_DISCOVERY_DELAY,
+            this.listenerJob = scheduledExecutorService.schedule(listenerRunnable, BACKGROUND_DISCOVERY_DELAY,
                     TimeUnit.SECONDS);
         }
     }
 
     private void cancelListenerJob() {
-        if (listenerJob != null) {
-            logger.debug("Canceling discovery listener job");
-            listenerJob.cancel(true);
+        logger.debug("Canceling discovery listener job");
```

Comment:
Why log this here when the discovery listener won't be canceled if listenerJob is not null?

Reply:
Due to the introduction of a local var this line ended up on the wrong side. Fixed.